### PR TITLE
Allow -port <num> or irc.host.tld <num> in /server add 

### PR DESCRIPTION
**Pull request by vague666**
Also in /server modify
irc.host.tld &lt;num&gt; has preference if both are specified

### CODE REVIEW

File: src/fe-common/core/fe-server.c [hunk 1]
```diff
@@ -117,7 +117,11 @@ static void cmd_server_add_modify(const char *data, gboolean add)
 		return;
 
 	if (*addr == '\0') cmd_param_error(CMDERR_NOT_ENOUGH_PARAMS);
-	port = *portstr == '\0' ? DEFAULT_SERVER_ADD_PORT : atoi(portstr);
+
+        value = g_hash_table_lookup(optlist, "port");
+        port = *portstr == '\0' ?
+                (value != NULL && *value != '\0' ? atoi(value) : DEFAULT_SERVER_ADD_PORT)
```
Comment (dequis) [inline src/fe-common/core/fe-server.c]: spaces instead of tabs

**Comment by dequis** [comment]
Other than the indentation nit, sounds good.

**Comment by ailin-nemui** [comment]
afaics this change will unconditionally modify the port even if you do not specify any port, can you double check?

**Comment by dequis** [comment]
Hm yeah, so something like changing `DEFAULT_SERVER_ADD_PORT` into `rec ? rec->port : DEFAULT_SERVER_ADD_PORT`. But that line has so many ternaries it's better to turn them into proper if blocks at this point.

**Comment by ailin-nemui** [comment]
can you change the ?:?: to a if, it's too unreadable for me

File: src/fe-common/core/fe-server.c [hunk 2]
```diff
@@ -117,7 +117,17 @@ static void cmd_server_add_modify(const char *data, gboolean add)
...
 	if (*addr == '\0') cmd_param_error(CMDERR_NOT_ENOUGH_PARAMS);
-	port = *portstr == '\0' ? DEFAULT_SERVER_ADD_PORT : atoi(portstr);
+
+	value = g_hash_table_lookup(optlist, "port");
+
+	if (*portstr != '\0')
+		port = atoi(portstr);
+	else if (value != NULL && *value != '\0')
+		port = atoi(value);
+	else if (g_hash_table_lookup(optlist, "tls"))
```
Comment (ailin-nemui) [inline src/fe-common/core/fe-server.c]: does we need to add "ssl" also for backward compat?

**Comment by ailin-nemui** [comment]
note to self: test the case of modifying a port and if irssi can handle multiple ports of the same host

**Comment by vague666** [comment]
@ailin-nemui, how did that work? Was it possible to have several servers with the same name but different ports? That's something I haven't tested, but I've tested all scenarios I could think of when it comes to single servers

**Comment by vague666** [comment]
There should be a test document that describes a set of operations that should be verified to work if something is changed. It's a big job to create such a document though

**Comment by ailin-nemui** [comment]
@vague666 in the past the idea was that you can have the same named server, with different ports, eg. (A) irc.example.net 6660, (B) irc.example.net 6661, and so on, and then you could modify the port of (A) with /server add -port 6662 irc.example.net 6660

but idk if it ever worked (I currently think it didn't)

**Comment by vague666** [comment]
I can't add the same server with another port in 1.0.2

**Pull request merged**